tests/src/test/scala/cats/tests/VectorSuite.scala

```diff
@@ -67,6 +67,30 @@ class VectorSuite extends CatsSuite {
     assert(Vector.empty[Int].toNev == None)
   }
 
+  test("groupByNev should be consistent with groupBy")(
+    forAll { (fa: Vector[Int], f: Int => Int) =>
+      assert((fa.groupByNev(f).map { case (k, v) => (k, v.toVector) }: Map[Int, Vector[Int]]) === fa.groupBy(f))
+    }
+  )
+
+  test("groupByNevA should be consistent with groupByNev")(
+    forAll { (fa: Vector[Int], f: Int => Int) =>
+      assert(fa.groupByNevA(f.andThen(Option(_))) === Option(fa.groupByNev(f)))
+    }
+  )
+
+  test("scanLeftNev should be consistent with scanLeft")(
+    forAll { (fa: Vector[Int], b: Int, f: (Int, Int) => Int) =>
+      assert(fa.scanLeftNev(b)(f).toVector === fa.scanLeft(b)(f))
+    }
+  )
+
+  test("scanRightNev should be consistent with scanRight")(
+    forAll { (fa: Vector[Int], b: Int, f: (Int, Int) => Int) =>
+      assert(fa.scanRightNev(b)(f).toVector === fa.scanRight(b)(f))
+    }
+  )
```

Comment:
These two tests do not exist for `ListOps`'s `scanLeftNel` and `scanRightNel` methods actually.
I wonder, does it make sense to add similar tests to `ListSuite` too?
It might be beyond of the PR's scope though...

Reply:
I took @LukaJCB's reaction as encouragement to do this, so I added missing tests to `ListSuite`.